clap_derive/src/derives/arg_enum.rs

```diff
@@ -86,6 +82,9 @@ fn lits(
             if let Kind::Skip(_) = &*attrs.kind() {
                 None
             } else {
+                if !matches!(variant.fields, Fields::Unit) {
+                    abort_call_site!("`#[derive(ArgEnum)]` only supports non-unit variants, unless they are skipped");
+                }
```

Comment:
Can you attach this error to the variant's span so the error message points exactly at this one?